### ISSUE TITLE
auth: returns SERVFAIL responses for presigned zones with NSEC3 enabled.

### ISSUE THREAD

**Issue by gbxyz** [PowerDNS/pdns#9263]
- Program: Authoritative
 - Issue type: Bug report

### Short description
For a pre-signed zone that uses NSEC3, PowerDNS (using the generic MySQL backend) answers queries for non-existent names with SERVFAIL.

### Environment
 - Operating system: CentOS 7
 - Software version: 4.2, 4.3, master
 - Software source: PowerDNS repository

### Steps to reproduce
1. import a pre-signed zone into the database using `zone2sql`. This omits the NSEC3 records as expected.
2. run `pdnsutil set-presigned ZONE` and `pdnsutil set-nsec3 "PARAMS"`, setting `PARAMS` to the content of the NSEC3PARAM record in the zone.
3. send a DNS query (with the EDNS DO header bit set) to PowerDNS for a non-existent name/type within the zone.

### Expected behaviour
PowerDNS should return a response with with an NXDOMAIN rcode, (or with NOERROR but no answer data), plus the usual SOA, NSEC3 and RRSIG records you'd expect in a negative answer.

### Actual behaviour
PowerDNS responds with a SERVFAIL.  When run with `loglevel=9` in the config file, the following exception is emitted:

```
Jun 19 11:29:41 c7 pdns_server: Exception building answer packet for does-not-exist.example.com/SOA (Attempt to print an unset dnsname) sending out servfail
```

### Other information
The SERVFAIL only happens after Step 3 (setting NSEC3 mode on the zone). Before then, PowerDNS answers with a synthesised NSEC record as expected based on what's in the documentation.

I am unable to check if the issue is due to the zone not being rectified, as running `pdnsutil rectify-zone` produces this error:

```
Rectify presigned zone 'example.com' is not allowed/necessary.
```

I am unable to share the zone file here but can provide it by email.

**Comment by Habbie** [comment]
Can you try `pdnsutil load-zone` instead of `zone2sql` please?

If that fails, can you try:

1. unset presigned
2. rectify
3. set presigned

I'm unsure if we even expect this to work - most users get their presigned zones via AXFR!

**Comment by gbxyz** [comment]
Neither of the above work, I still get the SERVFAIL.

**Comment by brynhood** [comment]
any update on this one?

**Comment by klaus-nicat** [comment]
When setting the NSEC3 option for a zone, the zone must get rectified. Hence, ordername must be set with the hashes. Look into your records table. If the ordername column does not have the hashes, then the reason for SERVFAIL is obvious. Then you have to find out why the zone does not get rectified.

Try to other way round:
1. Import Zone
2. Set NSEC3
3. Rectify
4. Set Presigned

Or: just AXFR the zone from some name server. 

btw: why do you import a presigned file? Who signs the presigned file? Or use for example the bindbackend to directly read the signed zone file

**Comment by Habbie** [comment]
This turns out to be a duplicate of #8892, so I'm closing it.
